benchmarks/pytorch-lightning/utils.py

```diff
@@ -0,0 +1,134 @@
+""" Utility functions / classes / stuffs """
+from __future__ import annotations
+
+import inspect
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Callable, NewType, TypeVar
+
+import pl_bolts.datamodules
+import torch
+from pl_bolts.datamodules import CIFAR10DataModule
+from pl_bolts.datamodules.vision_datamodule import VisionDataModule
+from simple_parsing import choice
+from simple_parsing.helpers.serialization.serializable import Serializable
+from torch import nn
+from torchvision import models
+
+C = NewType("C", int)
+H = NewType("H", int)
+W = NewType("W", int)
+ModuleType = TypeVar("ModuleType", bound=nn.Module)
+
+
+# TODO maybe use this so we don't have to download the datasets, or use milatools/milavision.
+torchvision_dir: Path | None = Path("/network/datasets/torchvision")
+if not torchvision_dir.exists():
+    torchvision_dir = None
+
+DEFAULT_DATA_DIR: Path = Path(
+    os.environ.get("SLURM_TMPDIR", os.environ.get("SCRATCH", "data"))
+)
```

Comment:
Not sure if this is used, but this is supposed to run anywhere (but typically from a Docker image), so no references to `/network`, `$SCRATCH` or `$SLURM_TMPDIR` can be made. The data has to live in a subdirectory of `$MILABENCH_DIR_DATA` and it should be downloaded there by the `prepare.py` script.

Reply:
Fixed in 78f6867